pkg/controllers/nodeclaim/disruption/drift.go

```diff
@@ -115,12 +115,19 @@ func (d *Drift) isDrifted(ctx context.Context, nodePool *v1beta1.NodePool, nodeC
 	return driftedReason, nil
 }
 
-// Eligible fields for static drift are described in the docs
+// Eligible fields for drift are described in the docs
 // https://karpenter.sh/docs/concepts/deprovisioning/#drift
 func areStaticFieldsDrifted(nodePool *v1beta1.NodePool, nodeClaim *v1beta1.NodeClaim) cloudprovider.DriftReason {
 	nodePoolHash, foundHashNodePool := nodePool.Annotations[v1beta1.NodePoolHashAnnotationKey]
+	nodePoolVersionHash, foundVersionHashNodePool := nodePool.Annotations[v1beta1.NodePoolHashVersionAnnotationKey]
 	nodeClaimHash, foundHashNodeClaim := nodeClaim.Annotations[v1beta1.NodePoolHashAnnotationKey]
-	if !foundHashNodePool || !foundHashNodeClaim {
+	nodeClaimVersionHash, foundVersionHashNodeClaim := nodeClaim.Annotations[v1beta1.NodePoolHashVersionAnnotationKey]
+
+	if !foundHashNodePool || !foundHashNodeClaim || !foundVersionHashNodePool || !foundVersionHashNodeClaim {
```

Comment:
Just to confirm: This logic means that we have a version where we propagate the hash version on but don't change any hash logic. That will allow the rollback scenario to work for users that decide later after they upgrade that they now want to go back to a previous version with a different hash version

Reply:
Yes, that will be exactly how the rollback scenario will work

Reply:
Before this commit, NodeClaims and NodePools won't have this hash version key, which means we won't be considering these nodes. Can we just change it to only check for the hash here? 
```suggestion
	if !foundHashNodePool || !foundHashNodeClaim {
```

Reply:
The hash controller will update the both NodePool and NodeClaim to add the `nodepool-hash-version`